executor/src/witgen/machines/block_machine.rs

```diff
@@ -153,13 +153,37 @@ impl<T: FieldElement> Machine<T> for BlockMachine<T> {
     fn witness_col_values(&mut self, fixed_data: &FixedData<T>) -> HashMap<String, Vec<T>> {
         std::mem::take(&mut self.data)
             .into_iter()
-            .map(|(id, values)| {
-                let mut values = values
+            .map(|(id, mut values)| {
+
+                // For all constraints to be satisfied, unused cells have to be filled with valid values.
+                // We do this, we construct a default block, by repeating the first input to the block machine.
+
+                if values.len() < 2 * self.block_size {
+                    log::warn!("Block machine was never used. Assuming that filling witness columns with zeros satisfies the constraints.");
```

Comment:
I think warn might be too high of a verbosity. It will always be shown and maybe "regular" users don't know what to do with that message.

Reply:
Yeah but maybe it's ok. Maybe something that gives the impression "hey something is weird, i'll try but if it fails this is likely the reason."

Reply:
OK, rephrased it a bit!